### PR TITLE
Added error boundaries to CodeEditor and layout component

### PR DESCRIPTION
**Pull request by ishankbahl**
Fix issue #3 , https://github.com/facebook/react/issues/11008 by adding error boundaries to CodeEditor and layout components. Nudging (using alert()) users to whitelist the site in AdBlocker in the alternative UI.

### CODE REVIEW

File: src/components/ErrorBoundary/ErrorBoundary.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        }
+
+        componentDidCatch(error) {
+            this.setState({ error });
+        }
+
+        render() {
+            console.log(this.state);
+            if (this.state.error) {
+                alert('Error, try to whitelist the site in AdBlocker/Cookie Whitelist');
```
Comment (bvaughn) [inline src/components/ErrorBoundary/ErrorBoundary.js]: I don't think this is the right way to warn people about this. I think a `console.error` log would be reasonable though. Maybe something like:

> If you're using an ad-blocker or have disabled cookies,
> Consider adding this website to the whitelist.

We should also keep in mind that not _all_ errors are caused by this so our wording shouldn't be overly assertive that this is definitely the cause/fix. Users might see an error even if they aren't using adblockers and haven't disabled cookies, and that would be confusing.

File: src/components/CodeEditor/CodeEditor.js [hunk 2]
```diff
@@ -17,6 +17,7 @@ import Remarkable from 'remarkable';
 import {LiveProvider, LiveEditor} from '@gaearon/react-live';
 import {colors, media} from 'theme';
 import MetaTitle from 'templates/components/MetaTitle';
+import ErrorBoundary from '../ErrorBoundary';
```
Comment (bvaughn) [inline src/components/CodeEditor/CodeEditor.js]: nit
```js
import ErrorBoundary from 'components/ErrorBoundary';
```

File: src/layouts/index.js [hunk 3]
```diff
@@ -25,6 +25,7 @@ import '../prism-styles';
 import 'glamor/reset';
 import 'css/reset.css';
 import 'css/algolia.css';
+import ErrorBoundary from '../components/ErrorBoundary';
```
Comment (bvaughn) [inline src/layouts/index.js]: nit
```js
import ErrorBoundary from 'components/ErrorBoundary';
```

File: src/components/ErrorBoundary/ErrorBoundary.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+                error: null,
+            };
+        }
+
+        componentDidCatch(error) {
+            this.setState({ error });
+        }
+
+        render() {
+            console.log(this.state);
```
Comment (bvaughn) [inline src/components/ErrorBoundary/ErrorBoundary.js]: Remove this^ 😄

File: src/layouts/index.js [hunk 5]
```diff
@@ -82,4 +83,4 @@ class Template extends Component {
   }
 }
 
-export default Template;
+export default ErrorBoundary(Template);
```
Comment (bvaughn) [inline src/layouts/index.js]: Errors in the template will actually result in a totally empty/white page. This is probably not great. We should show _something_ as a fallback in the browser (in addition to logging error info to the console).

Comment (ishankbahl) [reply src/layouts/index.js]: What do you suggest as the fallback?
Maybe we can use a modal (using react-modal) here.

Comment (bvaughn) [reply src/layouts/index.js]: We probably need to show a basic error page. It doesn't have to look super pretty. @joecritch or I can pretty it up.

**Comment by bvaughn** [comment]
It's been a while since there was any movement on this PR, and the code it modifies has been changed by subsequent PRs. I'm going to close this for now- but I'm happy to re-review if the PR is updated. Please just ping me by name if there are more changes pushed!

**Pull request closed without merging**